autogoal/kb/_algorithm.py

```diff
@@ -319,6 +332,12 @@ def repr_method(self):
     def getattr_method(self, attr):
         return getattr(self.inner, attr)
 
+    @classmethod
```

Comment:
Can we add some kind of test (either as docstring or as a proper test in `/tests/core`) that illustrates why this change is necessary? I know it's after the fact, but the idea is if we reverse these changes, is there some fairly simple use case that breaks? Maybe it's not trivial to find one, but I'd like to at least make the thought experiment of trying to come up with one.

Reply:
Did not think of or found any use case that is broken by this change. I've searched for other uses of the `inspect.signature` method, but all of them were being used with the `run` method and not the `__init__`.  There is one in the demo though ([this line](https://github.com/autogoal/autogoal/blob/136cb8c9a41ca59fafad56ee8a22a3c888a99b74/autogoal/contrib/streamlit/demo.py#L320)) but i think this is not going to break it.

Reply:
Do you think the docstring test is enough @apiad ?